File: leetcode/1846/solution.go
/**
 * https://leetcode.com/problems/maximum-element-after-decreasing-and-rearranging/
 * 1846. Maximum Element After Decreasing and Rearranging
 */

package main

import "sort"

func maximumElementAfterDecrementingAndRearranging(arr []int) int {
	sort.Ints(arr)
	answer := 1
	for i := 1; i < len(arr); i++ {
		if answer < arr[i] {
			answer += 1
		}
	}
	return answer
}

File: leetcode/1846/solution.py
"""
https://leetcode.com/problems/maximum-element-after-decreasing-and-rearranging/
1864. Maximum Element After Decreasing and Rearranging
"""


class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        arr.sort()
        answer = 1
        for i in range(1, len(arr)):
            answer = min(answer + 1, arr[i])
        return answer
